File: src/ClientModel/ClientResponse.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace AutoRest.Java.Model
{
    public sealed class ClientResponse
    {
        public string Name { get; }
        public string Package { get; }
        public string Description { get; }
        public IType HeadersType { get; }
        public IType BodyType { get; }

        public ClientResponse(string name, string package, string description, IType headersType, IType bodyType)
        {
            Name = name;
            Package = package;
            Description = description;
            HeadersType = headersType ?? PrimitiveType.Void;
            BodyType = bodyType ?? PrimitiveType.Void;
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/model/clientmodel/ClientResponse.java
package com.azure.autorest.model.clientmodel;

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

/** 
 The response that is returned by a ClientMethod.
*/
public final class ClientResponse
{
    private String Name;
    public String getName()
    {
        return Name;
    }
    private String Package;
    public String getPackage()
    {
        return Package;
    }
    private String Description;
    public String getDescription()
    {
        return Description;
    }
    private IType HeadersType;
    public IType getHeadersType()
    {
        return HeadersType;
    }
    private IType BodyType;
    public IType getBodyType()
    {
        return BodyType;
    }

    public ClientResponse(String name, String package_Keyword, String description, IType headersType, IType bodyType)
    {
        Name = name;
        Package = package_Keyword;
        Description = description;
        HeadersType = headersType;
        BodyType = bodyType;
    }
}